### PR TITLE
Bugfix: make ondemand feature be allowed to passed into as an object

### PR DESCRIPTION
**Pull request by aolfat**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: metadata/metadata.go [hunk 1]
```diff
@@ -715,15 +715,26 @@ func (resource *featureVariantResource) IsEquivalent(other ResourceVariant) (boo
 	thisProto := resource.serialized
 	otherProto := otherVariant.serialized
 
+	fmt.Printf("thisProto: %v\n", thisProto)
```
Comment (ahmadnazeri) [inline metadata/metadata.go:718]: remove these?

Comment (anthonylasso) [inline metadata/metadata.go:718]: still need these?

File: metadata/metadata.go [hunk 2]
```diff
@@ -715,15 +715,26 @@ func (resource *featureVariantResource) IsEquivalent(other ResourceVariant) (boo
...
+	fmt.Printf("otherProto: %v\n", otherProto)
+
 	isEquivalentLocation := false
 	if thisProto.GetFunction() != nil {
 		isEquivalentLocation = proto.Equal(thisProto.GetFunction(), otherProto.GetFunction())
 	} else {
 		isEquivalentLocation = proto.Equal(thisProto.GetColumns(), otherProto.GetColumns())
 	}
 
+	fmt.Printf("thisProto.GetFunction(): %v\n", thisProto.GetFunction())
```
Comment (ahmadnazeri) [inline metadata/metadata.go:728]: remove the print statements? or do we need them to be logs?

**Review by anthonylasso — APPROVED** [review]

File: metadata/metadata.go (hunk 1 above)
Comment (aolfat) [reply metadata/metadata.go:718]: ahhhh how did these get in there

**Review by ahmadnazeri — APPROVED** [review]

**Pull request merged**